pallets/account-linker/src/lib.rs

```diff
@@ -228,6 +240,52 @@ pub mod pallet {
 			Ok(().into())
 
 		}
+
+		#[pallet::weight(T::WeightInfo::link_polkadot())]
+		pub fn link_polkadot(
+			origin: OriginFor<T>,
+			account: T::AccountId,
+			index: u32,
+		) -> DispatchResultWithPostInfo {
+
+			let origin = ensure_signed(origin)?;
+
+			// TODO: charge some fee
+
+			<PolkadotPending<T>>::insert(origin, (account, index));
+
+			Ok(().into())
+
+		}
+
+		#[pallet::weight(T::WeightInfo::accept_polkadot())]
+		pub fn accept_polkadot(
+			origin: OriginFor<T>,
+			account: T::AccountId,
+		) -> DispatchResultWithPostInfo {
+
+			let origin = ensure_signed(origin)?;
+
+			let (target, index) = Self::polkadot_pending(&account);
+			ensure!(target == origin, Error::<T>::WrongPendingRequest);
```

Comment:
If there is currently no pending requests for this account, will WrongPendingRequest also be returned? This might be a bit misleading.
Maybe better to do empty check for the return value of this getter function.

Reply:
It's a bit tricky here since when the map is empty, it will return the default value, so target == origin will not pass.
Added a new test for it and you may check it: test_no_polkadot_pending_address